lib/sedna/dataset/dataset.py

```diff
@@ -18,71 +18,53 @@
 choice 2: a high level Dataset object not compatible with tensorflow,
 but it's unified in our framework.
 """
-
-import fileinput
 import logging
 import os
 
-import numpy as np
-
 from sedna.common.config import BaseConfig
 
 LOG = logging.getLogger(__name__)
 
 
-def _load_dataset(dataset_url, format, **kwargs):
+def _load_dataset(dataset_url, format, preprocess_fun, **kwargs):
     if dataset_url is None:
         LOG.warning(f'dataset_url is None, please check the url.')
         return None
     if format == 'txt':
         LOG.info(
             f"dataset format is txt, now loading txt from [{dataset_url}]")
-        if kwargs.get('with_image'):
-            return _load_txt_dataset_with_image(dataset_url)
+        samples = _load_txt_dataset(dataset_url)
+        if preprocess_fun:
+            new_samples = [preprocess_fun(s) for s in samples]
         else:
-            return _load_txt_dataset(dataset_url)
+            new_samples = samples
+        return new_samples
 
 
-def load_train_dataset(data_format, **kwargs):
+def load_train_dataset(data_format, preprocess_fun, **kwargs):
```

Comment:
this adds a extra preprocess_fun arg, how about code example's compatibility? such as the incremental learning example.

Reply:
In the incremental learning example,  it works fine with keeping preprocess_fun empty.